src/controllers/hello-world.ts

```diff
@@ -4,16 +4,24 @@
 // License text available at https://opensource.org/licenses/MIT
 'use strict';
 
-import {api, inject} from '@loopback/core';
-import {controllerSpec} from './hello-world.api';
-import {authenticate, AuthenticationBindings, UserProfile} from '@loopback/authentication';
+import { api, inject } from '@loopback/core';
+import { helloWorldApi } from '../apidefs/hello.world.api';
+import { authenticate, AuthenticationBindings, UserProfile } from '@loopback/authentication';
+import { UserRepository } from '../repositories/user.repository';
+import { User } from '../models/user';
 
-@api(controllerSpec)
+@api(helloWorldApi)
 export class HelloWorldController {
+  @inject('repositories.UserRepository') repository: UserRepository
   constructor(@inject(AuthenticationBindings.CURRENT_USER) private user: UserProfile) {}
 
   @authenticate('BasicStrategy')
-  helloWorld(name: string) {
-      return `Hello world ${name} ` + JSON.stringify(this.user);
+  async helloWorld(name?: string) {
+    await this.repository.create({ username: name });
```

Comment:
I find this rather silly, why would a `GET  /helloworld` request create a new user? GET requests must not be changing application state!

Please create a new endpoint for creating new users.

Reply:
I know it is unorthodox to do this, but I was thinking of persisting the user info provided in this endpoint, so that users can go to `/users` and see who has visited `/helloworld`.

Reply:
I agree with @bajtos that it violates how REST works.  Perhaps we could put it in the following way? 
GET /helloworld - to get the `hello world` response with the input name
GET /logs - to get a log of the inputs were used to call the /helloworld